### PR TITLE
Add navigation commands "Go to first / last parent commit"

### LINKED ISSUE #11863: Add navigation command "Go to last parent commit"
### Feature description

Add navigation command "Go to last parent commit" with hotkey `Ctrl+Right`
Rename "Go to parent commit" into "Go to first parent commit" with alternative hotkey `Ctrl+Left`

Before

grafik

grafik

### Environment

vNext

### PR DESCRIPTION
**Pull request by mstv**
Fixes #11863

## Proposed changes

- Add navigation commands `GoToFirstParent` / `GoToLastParent` to the RevisionGridControl and to the FileStatusList of the RevisionGridControl
  with hotkeys `Ctrl+Left` / `Ctrl+Right`
- Add icon to `GoToCommonAncestor`

I use this for following the main branch or for entering the feature branch, respectively.
Note: Left and right do not always match the graph - but strictly the merge direction.

`GoToFirstParent` is almost the same as `GoToParent`. But when moving down and up in branches, `GoToParent` chooses the parent commit from the navigation history.

### Before

grafik

grafik

### After

image

image

## Test methodology 

- manual

## Merge strategy

I agree that the maintainer squash merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Review by gerhardol — COMMENTED** [review]
The implementation is good.
What I have against this is the default use of two useful shortcuts for something I see very little use for myself. Convince me that this is useful.

**Comment by mstv** [comment]
> What I have against this is the default use of two useful shortcuts for something I see very little use for myself. Convince me that this is useful.

>> I use this for following the main branch or for entering the feature branch, respectively.

As an integrator, I often need to scan the history of parallel branches, and as reviewer, I need to find examples of nasty diffs. ;-)
We can also omit default hotkeys or choose other ones. I can assign my preferred ones manually.

File: src/app/GitUI/Hotkey/HotkeySettingsManager.cs [hunk 1]
```diff
@@ -289,6 +289,8 @@ public IReadOnlyList<HotkeySettings> CreateDefaultSettings()
                     Hk(RevisionGridControl.Command.DeleteRef, Keys.Delete),
                     Hk(RevisionGridControl.Command.GoToChild, Keys.Control | Keys.N),
                     Hk(RevisionGridControl.Command.GoToCommit, Keys.Control | Keys.Shift | Keys.G),
+                    Hk(RevisionGridControl.Command.GoToFirstParent, Keys.Control | Keys.Left),
+                    Hk(RevisionGridControl.Command.GoToLastParent, Keys.Control | Keys.Right),
```
Comment (gerhardol) [inline src/app/GitUI/Hotkey/HotkeySettingsManager.cs:293]: Approving, excluding the assign of theses keys (same for RevDiff) - at least for now.

**Review by gerhardol — APPROVED** [review]

File: src/app/GitUI/Hotkey/HotkeySettingsManager.cs (hunk 1 above)
Comment (mstv) [reply src/app/GitUI/Hotkey/HotkeySettingsManager.cs]: `Ctrl+Left` and `Ctrl+Right` are text-editing hotkeys. That's why I have limited the hotkeys to the revision grid and to the file list.
I agree that there could come up a more important function for these hotkeys in these areas, too.
On the hand, they have never been used.
Removed.

**Review by RussKie — APPROVED** [review]

**Pull request merged**